espresso/environment/espresso_caff_node_test.go

```diff
@@ -0,0 +1,266 @@
+		return nil, ErrorFailedToParseSequencerPort{Have: sequencerHostAndPort}
+	}
+
+	u := url.URL{
+		Scheme: "http",
+		Host:   net.JoinHostPort("localhost", sequencerPort),
+		Path:   "/",
+	}
+
+	// Let's start the Caff Node now.
+	// Configure our caff-node geth instance
+	caffNodeGeth, err := geth.InitL2(RoleCaffNode, system.L2GenesisCfg, system.Cfg.JWTFilePath)
+	if have, want := err, error(nil); have != want {
+		return nil, ErrorFailedToStartCaffNodeGeth{Cause: have}
+	}
+
+	// start our caff-node geth instance
+	if have, want := caffNodeGeth.Node.Start(), error(nil); have != want {
+		return nil, ErrorFailedToStartCaffNodeGeth{Cause: have}
+	}
+
+	system.EthInstances[RoleCaffNode] = caffNodeGeth
+	system.Cfg.Loggers[RoleCaffNode] = testlog.Logger(t, slog.LevelInfo).New("role", RoleCaffNode)
+
+	// Make a copy
+
+	caffNodeConfig := *system.Cfg.Nodes[e2esys.RoleVerif]
+	caffNodeConfig.Rollup = *system.RollupConfig
+	caffNodeConfig.Rollup.CaffNodeConfig = rollup.CaffNodeConfig{
+		IsCaffNode:                    true,
+		PollingHotShotPollingInterval: 30 * time.Millisecond,
+		HotShotUrls:                   []string{u.String()},
+	}
+
+	// Configure
+	e2esys.ConfigureL1(&caffNodeConfig, system.EthInstances[e2esys.RoleL1], system.L1BeaconEndpoint())
+	e2esys.ConfigureL2(&caffNodeConfig, caffNodeGeth, system.Cfg.JWTSecret)
+
+	// Create the Op Node Now
+	caffNodeConfig.Rollup.LogDescription(system.Cfg.Loggers[RoleCaffNode], chaincfg.L2ChainIDToNetworkDisplayName)
+	l := system.Cfg.Loggers[RoleCaffNode]
+
+	var opNodeError error
+	caffNode, err := opnode.NewOpnode(l, &caffNodeConfig, func(e error) {
+		opNodeError = e
+	})
+	if have, want := err, error(nil); have != want {
+		// Clean up the Caff Node Geth instance
+		caffNodeGeth.Close()
+		return nil, ErrorFailedToStartCaffNodeOpNode{Cause: have}
+	}
+
+	if have, want := opNodeError, error(nil); have != want {
+		caffNodeGeth.Close()
+		return nil, ErrorFailedToStartCaffNodeOpNode{Cause: have}
+	}
+
+	// Alright, we should have our Caff Node Launched now.
+
+	return &CaffNodeInstance{
+		OpNode: caffNode,
+		Geth:   caffNodeGeth,
+	}, nil
+}
+
+// TestE2eDevNetWithEspressoWithCaffNodeDeterministicDerivation is a test that
+// attempts to make sure that the caff node can derive the same state as the
+// original op-node (non caffeinated).
+//
+// This tests is designed to evaluate Test 3.1.1 as outlined within the
+// Espresso Celo Integration plan.  It has stated task definition as follows:
+//
+//	Arrange:
+//		Running Sequencer, Batcher in Espresso mode, Caff node  OP node.
+//		Balance of Alice is 0.
+//		Check that this is the case querying both Caff and OP nodes
+//	Act:
+//		Send a single transaction that transfers 1 coin to Alice
+//	Assert:
+//		Query the Caff node to check that Alice balance has been increased by 1
+//		Query the OP node to check that Alice balance has been increased by 1
+//
+// The actual tests is unable to make Alice's initial balance zero, and will
```

Comment:
I guess in this case one could just create a new address, but indeed the test is fine and even simpler this way.